airbyte-connector-builder-server/connector_builder/impl/default_api.py

```diff
@@ -2,21 +2,344 @@
 from connector_builder.generated.models.streams_list_request_body import StreamsListRequestBody
+from connector_builder.impl.low_code_cdk_adapter import LowCodeSourceAdapter
 from fastapi import Body
+from jsonschema import ValidationError
 
 
 class DefaultApiImpl(DefaultApi):
+    def __init__(self):
+        self.logger = logging.getLogger("airbyte.connector-builder")
+
     async def get_manifest_template(self) -> str:
         return "Hello World"
 
     async def list_streams(self, streams_list_request_body: StreamsListRequestBody = Body(None, description="")) -> StreamsListRead:
-        raise Exception("not yet implemented")
+        """
+        Takes in a low code manifest and a config to resolve the list of streams that are available for testing
+        :param streams_list_request_body: Input parameters to retrieve the list of available streams
+        :return: Stream objects made up of a stream name and the HTTP URL it will send requests to
+        """
+        adapter = LowCodeSourceAdapter(streams_list_request_body.manifest)
+
+        stream_list_read = []
+        for http_stream in adapter.get_http_streams(streams_list_request_body.config):
+            stream_list_read.append(
+                StreamsListReadStreams(
+                    name=http_stream.name,
+                    url=urljoin(http_stream.url_base, http_stream.path()),
+                )
+            )
+        return StreamsListRead(streams=stream_list_read)
 
     async def read_stream(self, stream_read_request_body: StreamReadRequestBody = Body(None, description="")) -> StreamRead:
-        raise Exception("not yet implemented")
+        """
+        Using the provided manifest and config, invokes a sync for the specified stream and returns groups of Airbyte messages
+        that are produced during the read operation
+        :param stream_read_request_body: Input parameters to trigger the read operation for a stream
+        :return: Airbyte record messages produced by the sync grouped by slice and page
+        """
+        return self.temp_read_stream(stream_read_request_body)
+
+    def temp_read_stream(self, stream_read_request_body: StreamReadRequestBody = Body(None, description="")) -> StreamRead:
+        try:
+            adapter = self._create_low_code_adapter(stream_read_request_body)
+        except ValidationError as error:
+            return KnownExceptionInfo(
+                message=f"Invalid connector manifest with error: {error.message}", exception_class_name=ValidationError.__name__
+            )
+
+        single_slice = StreamReadSlices(pages=[])
+        log_messages = []
+
+        first_page = True
+        current_records: List[object] = []
+        current_page_request: Optional[HttpRequest] = None
+        current_page_response: Optional[HttpResponse] = None
+        try:
+            # Message groups are partitioned according to when request log messages are received. Subsequent response log messages
```

Comment:
not necessarily for this PR, but it may pay off to encapsulate this component in a dedicated reader which clearly documents the expected format. I think all these if statements unnecessarily downplay certainty about the order of input we're receiving e.g: we know that the output will always have the format: 

```
{type: LOG, message: "request: ..."}
{type: LOG, message: "response: ..."}
{type: RECORD, ...}
...
{type: RECORD, ...}
# repeats
{type: LOG, message: "request: ..."}
etc..
```

This isn't very different from a serialization/deserialization problem. 

The serializer/deserializer class can then be used in CI such that if the format output by `read_records` changes for whatever reason (I accidentally log requests after response) it would fail the build

Reply:
That's a fair point that the bundle of if/elif could give that impression. When you say documents the expected format by creating a dedicate reader, I'd like to understand what you mean by that even if I don't do it in this PR. 

The way I understood your comment is that we have a `reader.get_groups(messages)` that returns a list of message groups? So that makes things more explicit about what constitutes a group of messages since the reader has to explicitly emit a grouped message. Whereas the current approach kind conflates parsing messages and groups altogether

Reply:
About documenting the expected format, I just mean that the structure of input we expect is: 
```
<log message containing the request>
<log message containing the response>
<a bunch of records>
```
repeated ad infinitum. This should be documented explicitly in a comment. 

In addition, I think it would be good to encapsulate this parsing complexity into its own component. It makes the parsing logic easier to test in isolation without accessing private methods, and makes the endpoint code easier to read since it doesn't in-line this non trivial parsing logic into the API endpoint impl directly and instead just does a `reader.get_groups` or something like that.